src/test/groovy/NotificationManagerStepTests.groovy

```diff
@@ -450,6 +450,145 @@ class NotificationManagerStepTests extends ApmBasePipelineTest {
     assertJobStatusSuccess()
   }
 
+  @Test
+  void test_notify_slack_with_multiple_channels() throws Exception {
+    def script = loadScript(scriptName)
+    try {
+      script.notifySlack(
+        build: readJSON(file: "build-info-manual.json"),
+        buildStatus: "SUCCESS",
+        changeSet: readJSON(file: "changeSet-info-manual.json"),
+        stepsErrors: readJSON(file: "steps-errors.json"),
+        testsErrors: readJSON(file: "tests-errors.json"),
+        testsSummary: readJSON(file: "tests-summary.json"),
+        channel: 'foo, bar, baaz',
+        credentialId: 'test',
+        enabled: true
+      )
+    }
+    catch(e) {
+      println e
+    }
+    printCallStack()
+    assertFalse(assertMethodCallContainsPattern('log', 'notifySlack: Error with the slack comment'))
+    assertTrue(assertMethodCallContainsPattern('slackSend', 'channel=foo'))
+    assertTrue(assertMethodCallContainsPattern('slackSend', 'channel=bar'))
+    assertTrue(assertMethodCallContainsPattern('slackSend', 'channel=baaz'))
+    assertJobStatusSuccess()
+  }
+
+  @Test
+  void test_notify_slack_with_multiple_wrong_channels() throws Exception {
+    def script = loadScript(scriptName)
+    try {
+      script.notifySlack(
+        build: readJSON(file: "build-info-manual.json"),
+        buildStatus: "SUCCESS",
+        changeSet: readJSON(file: "changeSet-info-manual.json"),
+        stepsErrors: readJSON(file: "steps-errors.json"),
+        testsErrors: readJSON(file: "tests-errors.json"),
+        testsSummary: readJSON(file: "tests-summary.json"),
+        channel: ',', // valid for the iterator but not for valid channels
+        credentialId: 'test',
+        enabled: true
+      )
+    }
+    catch(e) {
+      println e
+    }
+    printCallStack()
+    assertFalse(assertMethodCallContainsPattern('log', 'notifySlack: Error with the slack comment'))
```

Comment:
What do you think to check no calls to slackSend if empty?
```suggestion
    assertFalse(assertMethodCallContainsPattern('log', 'notifySlack: Error with the slack comment'))
    assertTrue(assertMethodCallOccurrences('slackSend', 0))
```

Reply:
Love it!